R/price_complete.R

```diff
@@ -38,11 +38,17 @@ price_complete <- function(product_id, date_min = NULL, date_max = NULL) {
     date_max <- max(df$date)
   }
 
-  dates <- seq.Date(date_min, date_max, by = "day")
+  if (date_min > min(df$date) || date_max < max(df$date)) {
```

Comment:
Hmmm. I'm not sure that the `||` is logically sound. What happens if one of these conditions is true but the other isn't, then it seems that the code in the conditional is doing the wrong thing.

How about this rather:

```
date_min <- max(date_min, min(df$date)
date_max <- min(date_max, max(df$date)

dates <- seq.Date(date_min, date_max, by = "day")
```

Please test though. I just sucked that out of my thumb. :)